### DISCUSSION TITLE
Which port do I need for ESP32D0WDQ6rev1 ?

### DISCUSSION THREAD

**Discussion by pn2new** [micropython/micropython#16257] [ESP32] [answered]
Somehow I can't figure out the port I need to download for an ESP32 board. esptool reports the chip is a ESP32D0WDQ6rev1. On the download page (https://micropython.org/download/?port=esp32), however, the selection is rather through selecting manufacturers and features...

Here are some photos of the board. It would also be great to know what type of OLED display this really is...
![IMG_20241116_145248](https://github.com/user-attachments/assets/89e9c75c-90b9-4b5e-900a-98959f8127db)
![IMG_20241116_145314](https://github.com/user-attachments/assets/c06fd12f-a808-4f46-be1a-4781111500c2)

**Comment by scruss** [comment, marked as answer]
Generic [ESP32](https://micropython.org/download/ESP32_GENERIC/) should work.

The display is very likely an SSD1306, which is well supported by MicroPython. The board looks a lot like the Lilygo OLED V2.0 TTGO, or a close clone of it. It's a few years old.

**Reply by pn2new** [reply]
Thanks, I got it working (but still need to work on the SSD).

**Reply by scruss** [reply]
I have one of these boards, somewhere. If you get very stuck I might be able to help, but I'm only vaguely aware of which bookcase this particular board may be hiding under.

The ssd1306 library is in micropython-lib: [ssd1306](https://github.com/micropython/micropython-lib/tree/01047889eb1acf115424fee293e03769f6916393/micropython/drivers/display/ssd1306): `mip install ssd1306`

**Reply by pn2new** [reply]
Thanks, in the meanwhile I also got the ssd working.
Do you by any chance have an schematic of the electronics? I wonder how exactly Vin, 5V and the USB connector are wired.

**Reply by scruss** [reply]
> Do you by any chance have an schematic of the electronics?

I don't, but it's not likely to be much different from the [LilyGO OLED v3.0](https://github.com/Xinyuan-LilyGO/T96-OLED-SSD1306) schematic. The older board uses a different LDO (AMS1117 vs ME6211; both 3.3 V output) and it doesn't have the battery connection. I wouldn't be surprised if 5V / Vbus / Vusb / Vin were all common, as there's no 5 V regulator on the board.

I still haven't found mine to check

**Reply by scruss** [reply]
Glad it works now. I found my board, and it's quite [different](https://xoxo.zone/@scruss/113550927112144117) from yours. Via [David Sullivan on Fosstodon](https://fosstodon.org/@tkroo/113551179557393937) I found mine is the same as was shipped with the [HackerBox 0044](https://www.instructables.com/HackerBox-0044-PCB-123/) kit, and that page has the crucial detail that the display has an Enable pin on Pin 16. So to make mine work I need:

```py3
p = Pin(16, Pin.OUT, value=1)
i2c = I2C(0, sda=4, scl=15)
oled = SSD1306_I2C(128, 64, i2c)
```
